admittance_controller/test/test_admittance_controller.cpp

```diff
@@ -24,8 +24,16 @@
 // Test on_configure returns ERROR when a required parameter is missing
 TEST_P(AdmittanceControllerTestParameterizedMissingParameters, one_parameter_is_missing)
 {
-  ASSERT_EQ(SetUpController(GetParam()), controller_interface::return_type::ERROR);
-  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_ERROR);
+  auto ret = SetUpController(GetParam());
+  // additionally, test params required during configure only if init worked
+  if (ret == controller_interface::return_type::OK)
+  {
+    ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_FAILURE);
```

Comment:
I created this test because it "happens" that `on_init` works but filters that have their own param declaration within the `on_configure` fails if one param is missing.
I can split it into 2 tests, but will required a similar test (not assert) of `on_init` passing first and then `on_configure` to fail if you think this is better.

Reply:
I think it is better. Having flow-control in tests makes them hard to understand.

Reply:
ok will work on that tomorrow

Reply:
Sorry for the delay, I have removed the conditional test, but a second test will be added in a separate commit when filters are involved, because there, the `on_configure` could fail after a successful `on_init`.